java/lance-namespace-glue/src/main/java/com/lancedb/lance/namespace/glue/GlueNamespace.java

```diff
@@ -74,9 +99,349 @@ public ListNamespacesResponse listNamespaces(ListNamespacesRequest request) {
+  }
+
+  private void validateParent(List<String> id, String instance) {
+    if (id != null && id.size() > 1) {
+      throw LanceNamespaceException.badRequest(
+          "Nested namespaces not supported",
+          "/errors/bad-request",
+          instance,
+          "Glue does not support nested namespaces. Found nested path: " + String.join("/", id));
+    }
+  }
+
+  private void validateNamespaceName(String name, String instance) {
+    if (name == null || name.isEmpty()) {
+      throw LanceNamespaceException.badRequest(
+          "Invalid namespace name",
+          "/errors/bad-request",
+          instance,
+          "Namespace name cannot be null or empty");
+    }
+  }
+
+  private String extractNamespaceName(List<String> id) {
+    if (id == null || id.isEmpty()) {
+      throw LanceNamespaceException.badRequest(
+          "Invalid namespace identifier",
+          "/errors/bad-request",
+          NAMESPACE_INSTANCE,
+          "Namespace identifier cannot be null or empty");
+    }
+    return id.get(0);
+  }
+
+  private String buildNamespaceInstance(String namespaceName) {
+    return namespaceName != null ? NAMESPACE_INSTANCE + namespaceName : NAMESPACE_INSTANCE;
+  }
+
+  private String formatGlueExceptionDetail(GlueException e) {
+    StringWriter sw = new StringWriter();
+    PrintWriter pw = new PrintWriter(sw);
+    pw.println(e.getClass().getSimpleName() + ": " + e.getMessage());
+    e.printStackTrace(pw);
+    return sw.toString();
+  }
+
+  private static Map<String, String> extractDatabaseProperties(Database database) {
+    Map<String, String> glueProperties = Maps.newHashMap(database.parameters());
+    if (database.locationUri() != null) {
+      glueProperties.put(GlueConstants.PARAM_LOCATION, database.locationUri());
+    }
+    if (database.description() != null) {
+      glueProperties.put(GlueConstants.PARAM_DESCRIPTION, database.description());
+    }
+    return glueProperties;
+  }
+
+  private boolean namespaceExists(String namespaceName, String instance) {
+    try {
+      glueClient.getDatabase(
+          GetDatabaseRequest.builder()
+              .catalogId(config.glueCatalogId())
+              .name(namespaceName)
+              .build());
+      return true;
+    } catch (EntityNotFoundException e) {
+      return false;
+    } catch (GlueException e) {
+      throw LanceNamespaceException.serverError(
+          "Failed to get Glue Database: " + namespaceName,
+          "/errors/glue-service-error",
+          instance,
+          formatGlueExceptionDetail(e));
+    }
+  }
+
+  private Database getDatabase(String namespaceName, String instance) {
+    try {
+      return glueClient
+          .getDatabase(
+              GetDatabaseRequest.builder()
+                  .catalogId(config.glueCatalogId())
+                  .name(namespaceName)
+                  .build())
+          .database();
+    } catch (EntityNotFoundException e) {
+      throw LanceNamespaceException.notFound(
```

Comment:
feels like we should have a GlueToLanceErrorConverter to centralize these conversion logics.